projects/optic/src/commands/diff/diff.ts

```diff
@@ -258,34 +286,54 @@ const getDiffAction =
     options: DiffActionOptions
   ) => {
     const files: [string | undefined, string | undefined] = [file1, file2];
-    let shouldExit1 = false;
+    let parsedFiles: [ParseResult, ParseResult];
     if (file1 && file2) {
-      const parsedFiles = await getBaseAndHeadFromFiles(file1, file2);
-      const diffResult = await runDiff(files, parsedFiles, config, options);
-      shouldExit1 = diffResult.checks.failed > 0 && options.check;
+      parsedFiles = await getBaseAndHeadFromFiles(file1, file2, config);
     } else if (file1) {
-      if (config.vcs !== VCS.Git) {
+      if (config.vcs?.type !== VCS.Git) {
         const commandVariant = `optic diff <file> --base <ref>`;
         console.error(
           `Error: ${commandVariant} must be called from a git repository.`
         );
         process.exitCode = 1;
         return;
       }
-      const parsedFiles = await getBaseAndHeadFromFileAndBase(
+      parsedFiles = await getBaseAndHeadFromFileAndBase(
         file1,
         options.base,
-        config.root
+        config.root,
+        config
       );
-      const diffResult = await runDiff(files, parsedFiles, config, options);
-      shouldExit1 = diffResult.checks.failed > 0 && options.check;
     } else {
       console.error(
         'Command removed: optic diff (no args) has been removed, please use optic diff <file_path> --base <base> instead'
       );
       process.exitCode = 1;
       return;
     }
+
+    const diffResult = await runDiff(files, parsedFiles, config, options);
+    if (config.isAuthenticated) {
+      const [baseParseResult, headParseResult] = parsedFiles;
+      const apiId: string | null = 'TODO'; // headParseResult.jsonLike[OPTIC_URL_KEY] ?? baseParseResult.jsonLike[OPTIC_URL_KEY] ?? null
+      const shouldUploadBaseSpec = baseParseResult.context && apiId;
+      const shouldUploadHeadSpec = headParseResult.context && apiId;
+      if (shouldUploadBaseSpec) {
+        // TODO upload spec
+      }
+      if (shouldUploadHeadSpec) {
+        // TODO upload spec
+      }
+
+      const shouldUploadResults =
+        (shouldUploadBaseSpec || baseParseResult.isEmptySpec) &&
+        (shouldUploadHeadSpec || headParseResult.isEmptySpec);
+
+      if (shouldUploadResults) {
+        // TODO upload results
+      }
+    }
```

Comment:
I'm going to implement this in a follow up PR but I want to make sure this logic makes sense.

We only upload if there's an API Id we can detect (this will be pulled from the optic-url, first from the head spec and then from the base spec if head spec does not have it (i.e. when spec is empty)).

We then only upload the specs if it is unchanged in git. We only upload the results if both the base + head are unchanged in git OR they are empty.

Reply:
> We then only upload the specs if it is unchanged in git. We only upload the results if both the base + head are unchanged in git OR they are empty.

I'm not sure I understand this. Why wouldn't we upload them if they're changed?

Reply:
Ah what I mean here is that the working directory is clean